config/routes.rb

```diff
@@ -4,11 +4,10 @@
   # Defines the root path route ("/")
   root "users#index"
 
-  shallow do
-    resources :users, only: [:index, :show] do
-      resources :posts, only: [:index, :show]
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :new, :create, :show] do
+      resources :comments, only: [:create]
```

Comment:
- [ ] You should have a route for new comments. Youre probabbly missing it because you did not have the new action in your comments controller. Kindly check and add. When I run `rails routes`, I cannot find it among the routes just to be very sure.
- The route should be a `GET`
-`GET /posts/:post_id/comments/new(.:format) comments#new`

Reply:
The task requires that we create a form to handle the creation of comments. It did not specifically ask that we create a view for it. So I put the form in the show view of the post controller. It will make it easy for the user to comment on a post easily without having to navigate to another page.